src/pages/login.js

```diff
@@ -0,0 +1,204 @@
+import CheckoutTopHat from "@reactioncommerce/components/CheckoutTopHat/v1";
+import GuestForm from "@reactioncommerce/components/GuestForm/v1";
+import ShopLogo from "@reactioncommerce/components/ShopLogo/v1";
+import withCart from "containers/cart/withCart";
+import Link from "components/Link";
+
+import ChevronLeftIcon from "mdi-material-ui/ChevronLeft";
+
+const flexWrapper = () => ({
+  alignItems: "stretch",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "flex-start"
+});
+
+const styles = (theme) => ({
+  loginWrapper: {
+    ...flexWrapper(theme),
+    paddingBottom: theme.spacing.unit * 8,
+    [theme.breakpoints.up("md")]: {
+      minHeight: "400px",
+      paddingBottom: 0,
+      paddingRight: theme.spacing.unit * 8
+    }
+  },
+  guestWrapper: {
+    ...flexWrapper(theme),
+    borderTop: `solid 1px ${theme.palette.reaction.black10}`,
+    paddingTop: theme.spacing.unit * 8,
+    [theme.breakpoints.up("md")]: {
+      borderLeft: `solid 1px ${theme.palette.reaction.black10}`,
+      borderTop: "none",
+      paddingLeft: theme.spacing.unit * 8,
+      paddingTop: 0
+    }
+  },
+  backLink: {
+    "color": theme.palette.reaction.black80,
+    "cursor": "pointer",
+    "fontFamily": theme.typography.fontFamily,
+    "fontSize": 14,
+    "&:hover": {
+      color: theme.palette.reaction.reactionBlue400
+    }
+  },
+  backLinkText: {
+    letterSpacing: "0.3px",
+    lineHeight: 1.71,
+    marginLeft: theme.spacing.unit,
+    textDecoration: "underline"
+  },
+  loginButton: {
+    marginTop: theme.spacing.unit * 3
+  },
+  headerCenter: {},
+  headerFlex: {
+    alignSelf: "center",
+    flex: "1 1 1%"
+  },
+  header: {
+    alignContent: "center",
+    borderBottom: `solid 1px ${theme.palette.reaction.black10}`,
+    display: "flex",
+    justifyContent: "center",
+    marginBottom: theme.spacing.unit * 3,
+    padding: theme.spacing.unit * 3
+  },
+  logo: {
+    color: theme.palette.reaction.reactionBlue,
+    margin: "auto",
+    borderBottom: `solid 5px ${theme.palette.reaction.reactionBlue200}`
+  },
+  main: {
+    flex: "1 1 auto",
+    maxWidth: theme.layout.mainLoginMaxWidth,
+    minHeight: "calc(100vh - 135px)",
+    margin: "0 auto",
+    padding: `${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px 0`,
+    [theme.breakpoints.up("md")]: {
+      padding: `${theme.spacing.unit * 10}px ${theme.spacing.unit * 3}px 0`
+    }
+  },
+  root: {}
+});
+
+@withCart
+@withStyles(styles, { withTheme: true })
+export default class Login extends Component {
+  static propTypes = {
+    cart: PropTypes.shape({
+      account: PropTypes.object,
+      email: PropTypes.string
+    }),
+    classes: PropTypes.object,
+    setEmailOnAnonymousCart: PropTypes.func,
+    shop: PropTypes.shape({
+      name: PropTypes.string.isRequired,
+      description: PropTypes.string
+    }),
+    theme: PropTypes.object.isRequired
+  };
+
+  static getDerivedStateFromProps({ cart }) {
```

Comment:
To prevent a "flash render", caused by  the component's render being called immediately after this method is executed, this login should be move to getInitialProps https://github.com/zeit/next.js/wiki/Redirecting-in-%60getInitialProps%60